firebase-common/src/main/java/com/google/firebase/provider/FirebaseInitProvider.java

```diff
@@ -27,11 +27,16 @@
 import androidx.annotation.VisibleForTesting;
 import com.google.android.gms.common.internal.Preconditions;
 import com.google.firebase.FirebaseApp;
+import com.google.firebase.StartupTime;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /** Initializes Firebase APIs at app startup time. */
 public class FirebaseInitProvider extends ContentProvider {
-
   private static final String TAG = "FirebaseInitProvider";
+  @Nullable
+  public static StartupTime startupTime = StartupTime.now();
+  @NonNull
+  public static AtomicBoolean currentlyInitializing = new AtomicBoolean(false);
```

Comment:
I think this could benefit from some encapsulation not to allow direct modification, might make sense to expose this as say `public static boolean isCurrentlyInitializing()` and making the atomic private, wdyt?